### PR TITLE
Version bump: app-misc/vcontrold-0.98.6

### PR DESCRIPTION
**Pull request by l3u**
A new version has been tagged recently. Can be built by simply renaming the existing ebuild. Compiles and works fine on my Raspberry Pi :-)

### CODE REVIEW

**Comment by l3u** [comment]
I hope the commit message is okay now?

File: app-misc/vcontrold/vcontrold-0.98.6.ebuild [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+EAPI=6
+
+inherit cmake-utils
+
+DESCRIPTION="Daemon for communication with Viessmann Vito heatings"
+HOMEPAGE="https://github.com/openv/vcontrold/"
+SRC_URI="https://github.com/openv/vcontrold/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
```
Comment (mgorny) [inline app-misc/vcontrold/vcontrold-0.98.6.ebuild]:
```
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
```

Isn't it GPL-3+?

Comment (l3u) [reply app-misc/vcontrold/vcontrold-0.98.6.ebuild]: You're right! I changed it for both ebuilds.

**Comment by mgorny** [comment]
You shouldn't really be modifying the old ebuild in a version bump commit.

**Comment by l3u** [comment]
Changed it. Removed the change from the old ebuild.

**Comment by l3u** [comment]
Sorry, I can't reproduce this … I can build it both on my amd64 machine and my Raspberry Pi without a problem! No error, no warning. What's that ninja stuff?! "A small build system similar to make"? I wrote the vcontrold cmake file myself, and I didn't use this for sure!

**Comment by l3u** [comment]
Okay, I looked into the sources. The documentation cmake file actually generates a gzipped manpage. I patched the gzipping away, but – as with the gzipped variant before, portage doesn't install the very file generated by the build system, but (in my case) a bzip2 compressed version, no matter if the cmake file installs a *.1.gz manpage or a *.1 one.

So nothing changes if I remove the gzipping. Still works as before and still produces the very same result here.

Should I really add a patch causing no change? Or what exactly should I do? Sorry for dumb questions, I simply don't know what I should change …

**Comment by mgorny** [comment]
New version of Portage complain verbosely when you install `.gz`. It should really be uncompressed for portability.

**Comment by l3u** [comment]
Here you go.

**Comment by l3u** [comment]
Also fixed it upstream btw., so that we won't need this patch with the next release anymore.

File: app-misc/vcontrold/files/man_generation.patch [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ 
+ foreach(MANUAL IN LISTS MANUALS)
+-    set(MANPAGE_OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${MANUAL}.1.gz)
++    set(MANPAGE_OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${MANUAL}.1)
+     set(MANPAGE_SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/${MANUAL}.rst)
+     add_custom_command(OUTPUT ${MANPAGE_OUTPUT}
+-        COMMAND ${RST2MAN} ${RST2MAN_OPTS} ${MANPAGE_SOURCE} | gzip > ${MANPAGE_OUTPUT}
++        COMMAND ${RST2MAN} ${RST2MAN_OPTS} ${MANPAGE_SOURCE} ${MANPAGE_OUTPUT}
+         DEPENDS ${MANPAGE_SOURCE}
+         BYPRODUCTS ${MANPAGE_OUTPUT}
```
Comment (mgorny) [inline app-misc/vcontrold/files/man_generation.patch]: I think this is what makes ninja unhappy. How can the same file be output and byproduct at the same time?

**Comment by l3u** [comment]
What about this one? I can't test or don't know how to test a "ninja" build … but I'll bne glad to push the fix upstream if this makes it work!

**Comment by mgorny** [comment]
Thanks. I'm going to test it right now, and let you know if it solved the problem.

**Comment by mgorny** [comment]
Yep, removing BYPRODUCTS fixes ninja problems.

Also:
```
RepoMan scours the neighborhood...
  ebuild.minorsyn               1
   app-misc/vcontrold/vcontrold-0.98.6.ebuild: Unquoted Variable on line: 24
RepoMan sez: "You're only giving me a partial QA payment?
              I'll take it this time, but I'm not happy."
```

I'll fix that for you and merge it.

**Comment by l3u** [comment]
Thanks for tracking this down (and fixing the quoting, I simply forgot to run repoman), I'll also fix it upstream :-)

**Pull request closed without merging**